### ISSUE TITLE
Add torch.no_grad() when in eval mode

### ISSUE THREAD

**Issue by George3d6** [mindsdb/lightwood#154]
Currently there's various points where we use networks just for the feedforward and we place them in eval mode but don't use `torch.no_grad()`. We should change this, since no_grad will speed up the feedforward by disabling autograd (initially I thought that would be done by eval, but apparently not).

**Comment by George3d6** [comment]
This has now been implemented in all places where it was required.
